Automattic/sensei
# Fix previous questions that have changed teachers

### Changes proposed in this Pull Request

* Introduces a stateful and simple batch background job base classes. This will massively simplify these types of jobs in the future and make it really easy to use.
* Adds a job that is enqueued when updating from pre-3.9.0 that attempts to fix questions.
* Updates PHPUnit to 7.5 to match WordPress core and get benefit of some PHP 7.4 fixes.

### Testing instructions

On `version/3.8.0` tag...
* As teacher A, add a course with 11 lessons + quizzes. Add one question to each quiz.
* As admin, change course teacher from teacher A to teacher B.
* Make sure your `sensei-version` option is less than 3.9.0.

Switch to this branch...
* Load any WP page.
* Observe a `sensei_background_job_stateful` event scheduled (in WP cron or ActionScheduler) with arg `class` of `Sensei_Update_Fix_Question_Author`.
* Observe it will batch through quizzes.
* Make sure all questions from are transferred to teacher B from the course created above.
